packages/boot/test/bootstrapTests/supports.js

```diff
@@ -136,51 +154,42 @@ export const makeRunUtils = (controller, log = (..._) => {}) => {
    * @type {typeof E & {
    *   sendOnly: (presence: unknown) => Record<string, (...args: any) => void>;
    *   vat: (name: string) => Record<string, (...args: any) => Promise<any>>;
-   *   rawBoot: Record<string, (...args: any) => Promise<any>>;
    * }}
    */
   // @ts-expect-error cast, approximate
   const EV = presence =>
     new Proxy(harden({}), {
-      get: (_t, methodName, _rx) =>
+      get: (_t, method, _rx) =>
         harden((...args) =>
-          runMethod('messageVatObject', [{ presence, methodName, args }]),
+          queueAndRun(() =>
+            controller.queueToVatObject(presence, method, args),
+          ),
```

Comment:
I think `queueAndRun` would be simpler if it accepted a kpid rather than a callback that returns a kpid.
```suggestion
          {
            const kpid = controller.queueToVatObject(presence, method, args);
            return queueAndRun(kpid);
          },
```
(and probably also a rename in that case, likely related to `awaitRun` in other files)

Reply:
I had the same thought when I first encountered the code this derives from -- it seemed like an obvious simplification.  When I tried it, it blew up in my face.  The purpose of the more complicated pattern is so that the `queueTo...` call happens inside `runThunk`, which waits on the mutex queue before allowing the code that enqueues the message to be executed.  This ensures that the modification of the kernel run queue isn't attempted while a `controller.run()` is in progress.  This is related to the hazard that `EV` introduces, namely that it pretends that message sends and `controller.run()` completions are 1:1 correlated, which _can_ happen and often does, but there's absolutely no guarantee.  There's nothing that couples the completion of `controller.run()` to the settling of the promises that prior `queueTo...` calls returned.  You're just betting that's how it goes; you can set up a test to have that happen, but the onus is on you to arrange for it.  This is very different from being inside a vat, where an `await` or a `.then` can delay indefinitely.